### PR TITLE
[CARBONDATA-1215][BUGFIX] Fix unsafe column page for decimal query

### PR DESCRIPTION
**Pull request by jackylk**
For decimal loading is using variable length column page, in unsafe column page implementation, there is a bug that the page is not checking for this capacity.

In this PR, capacity check is added, now `UnsafeVarLengthColumnPage` will grow its size based on need.

A testcase is added to verify this bug fix

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/246/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2826/

File: core/src/main/java/org/apache/carbondata/core/datastore/page/VarLengthColumnPageBase.java [hunk 1]
```diff
@@ -106,7 +106,7 @@ static ColumnPage newDecimalColumnPage(byte[] lvEncodedBytes) throws MemoryExcep
 
     VarLengthColumnPageBase page;
     if (unsafe) {
-      page = new UnsafeVarLengthColumnPage(DECIMAL, numRows);
+      page = new UnsafeVarLengthColumnPage(DECIMAL, numRows, lvEncodedBytes.length);
```
Comment (QiangCai) [inline core/src/main/java/org/apache/carbondata/core/datastore/page/VarLengthColumnPageBase.java]: how about to use offset variable?

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2829/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/249/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/257/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2837/

**Comment by QiangCai** [comment]
LGTM

**Pull request closed without merging**